plugin-server/src/utils/db/db.ts

```diff
@@ -673,109 +654,70 @@ export class DB {
+                }
+            } catch (error) {
+                captureException(error)
             }
-        })
 
-        // We update the cache for every row, including missing ones to avoid hitting the DB consistently
-        this.promiseManager.trackPromise(this.updateGroupDataCache(teamId, result))
-        return result
-    }
+            this.statsd?.increment('group_info_cache.miss')
 
-    public async fetchGroupColumnsValues(teamId: number, groups: GroupIdentifier[]): Promise<Record<string, string>> {
-        if (groups.length === 0) {
-            return {}
-        }
-        const cachedResults = await Promise.all(
-            groups.map((groupIdentifier) => this.getGroupDataCache(teamId, groupIdentifier))
-        )
+            // If we didn't find cached data, lookup the group from Postgres
+            const storedGroupData = await this.fetchGroup(teamId, groupTypeIndex as GroupTypeIndex, groupKey)
 
-        let columns: Record<string, string> = this.groupCacheDataToColumns(cachedResults)
-        const groupsWithoutCachedData = cachedResults
-            .filter((groupData) => !groupData.cached)
-            .map((groupData) => groupData.identifier)
+            if (storedGroupData) {
+                groupColumns[propertiesColumnName] = JSON.stringify(storedGroupData.group_properties)
 
-        if (groupsWithoutCachedData.length > 0) {
-            const fetchedResults = await this.fetchGroupDataAndUpdateCache(teamId, groupsWithoutCachedData)
-            columns = { ...columns, ...this.groupCacheDataToColumns(fetchedResults) }
-        }
-        return columns
-    }
+                const createdAt = castTimestampOrNow(storedGroupData.created_at)
+                groupColumns[createdAtColumnName] = createdAt
 
-    private groupCacheDataToColumns(groupCacheData: GroupCacheData[]): Record<string, string> {
-        const result: Record<string, string> = {}
-        for (const group of groupCacheData) {
-            if (group.data) {
-                result[`group${group.identifier.index}_properties`] = JSON.stringify(group.data.properties)
-                result[`group${group.identifier.index}_created_at`] = castTimestampOrNow(
-                    group.data.created_at,
-                    TimestampFormat.ClickHouse
-                )
+                // We found data in Postgres, so update the cache
+                // We also don't want to throw here, worst case is we'll have to fetch from Postgres again next time
+                try {
+                    await this.updateGroupCache(teamId, groupTypeIndex, groupKey, {
+                        properties: storedGroupData.group_properties,
+                        created_at: createdAt,
+                    })
+                } catch (error) {
+                    captureException(error)
+                }
+            } else {
+                // We couldn't find the data from the cache nor Postgres, so record this in a metric and in Sentry
```

Comment:
this can happen because we only create the group if `groupidentify` is called ... this makes me think should we change that and always create a group if it doesn't exist? And furthermore if we did that we could pass the group info forward similar to person info and not need to have caching here at all either?

Reply:
Yeah I think always creating like we do with persons is a good idea. Worth trying to find out why we didn't do this in the first place

Reply:
persons was changed recently with the changes to the pipeline, I think this is just historical.